### ISSUE TITLE
add R console output and other text stuff to HTML output

### ISSUE THREAD

**Issue by taylori** [r4ss/r4ss#76]
@allanhicks mentioned that @colemonnahan might be able to contribute something like this. Seems like a good idea.

Also, format of output to R console hasn't been changed in about 10 years. Probably could be cleaned up a bit.

**Comment by colemonnahan** [comment]
Allan was referring to me displaying the output live, not updating it. We don't see any of the output in ss3sim so I'm probably not the right person to update this.

Although I'll probably be adding some functionality for the EWAA models that Ian S wanted, so it's good I've got write access now.

**Comment by taylori** [comment]
Thanks to @ChristineStawitz-NOAA, three of the tables reported in the R console have been added to the HTML viewer created by `SS_plots` in a new `DiagnosticTables` tab shown below.

These include: 

- A table of parameters to help check for things hitting bounds or with high gradients.
- A table of pairs of parameters with high correlations (suggesting that they might be confounded and difficult to estimate simultaneously or cause issues with MCMC convergence). 
- A table of parameters which have low correlation with all other parameters (suggesting that they should be looked at to make sure they are contributing to the model)

In addition to being in a more convenient place than the R console, HTML allows useful color coding.

![image](https://user-images.githubusercontent.com/4992918/56328405-b2a69a00-6133-11e9-9696-c2313c2c3b62.png)

Additional elements of the R console output could be added to this tab in the future.

**Comment by colemonnahan** [comment]
To be slightly pedantic the linear correlations have no impact on the MCMC efficiency. They are negated by the mass matrix transformation of the posterior surface. It's good to report this information though, it's just a misunderstanding I see a lot.

**Comment by taylori** [comment]
@colemonnahan if the shape is truly linear that makes sense, but won't the relationship between the parameters often change as you move away from the MLE leading to lower efficiency for the traditional MCMC approaches (as in contrived example below from http://www.admb-project.org/developers/mcmc/presentation-on-mcmc-options.pdf). 

But thanks to you, we don't need to be using the standard MCMC in ADMB in the future.

![image](https://user-images.githubusercontent.com/4992918/56396622-93bc0c80-61f4-11e9-9e18-2bb06e8c0146.png).

**Comment by colemonnahan** [comment]
Correct, but the .cor matrix relates the linear correlation at the MLE, and this matrix is used to rotate the space. So if it's correct and the correlation is linear, then there will be no effect. But this does not always map to non-linear correlations as I'm sure you already know: https://en.wikipedia.org/wiki/Correlation_and_dependence#Correlation_and_linearity ... so looking at high correlations is not a reliable way to detect problematic non-linear correlations.

Plus in my practical experience it is not the source of problems. Look to parameters on bounds or those with huge or tiny standard errors.

**Comment by taylori** [comment]
Good points. Perhaps they aren't a source of problems in MCMC as I thought, but I think the high correlations can help identify ways to achieve a more parsimonious model.  

The table below shows strong correlations between two pairs of double-normal selectivity parameters. In one case it's parameters 1 and 3 (peak and the ascending slope) which you're stuck estimating. In the other case, it's parameters 4 and 6 controlling the descending slope and selectivity at the maximum size. Seeing this strong negative correlation in that second pair could be used as evidence that the model should be simplified by fixing parameter 6 at -999 to allow the right-hand side of the double normal to decay smoothly, which I think is a better model setup in general.

  | label.i | label.j | corr
-- | -- | -- | --
2 | SizeSel_22P_6_22_CA_Tri_late | SizeSel_22P_4_22_CA_Tri_late | -0.969
1 | SizeSel_4P_3_4_CA_NTWL | SizeSel_4P_1_4_CA_NTWL | 0.977

**Comment by colemonnahan** [comment]
I totally agree Ian. I just wanted to put that small technical detail out there since I see this misunderstanding commonly. This will be a useful addition to the output.

**Comment by iantaylor-NOAA** [comment]
Thanks @k-doering-NOAA for bring this back up. I think we can make this a higher priority to make progress on over the summer.
This issue got sidetracked in the discussion of which diagnostics are actually helpful, but I still agree with the underlying goal of putting all this into the HTML instead of expecting folks to read the R console output when running `SS_output()` and `SS_plots()`.

I'm also interested in working in the short term on the somewhat related #528 "add HTML view for SSplotComparisons()".

**Comment by iantaylor-NOAA** [comment]
I just tried to come up with a list of stuff to put into the HTML view and I'm not actually seeing a whole lot of useful stuff.
The biggest thing that I can think of would be info on the model tuning. Some of that is found interspersed between the plots of length and age comps, but it would be useful to get some quick feedback on whether the completed model is adequately tuned or not.

**Comment by k-doering-NOAA** [comment]
Yes, I agree model tuning info could be a great addition! Should it get its own tab, or be added to an existing one?

**Comment by k-doering-NOAA** [comment]
@iantaylor-NOAA I added you ass the assignee, as I think this is related to your idea of working on the HTML output before the next assessment cycle, perhaps. But it might be beyond the scope of what you are hoping to do. Feel free to unassign yourself!
